server/schedule/filter/filters.go

```diff
@@ -601,3 +601,72 @@ const (
 
 var allSpecialUses = []string{SpecialUseHotRegion, SpecialUseReserved}
 var allSpeicalEngines = []string{EngineTiFlash}
+
+type isolationFilter struct {
+	scope          string
+	locationLabels []string
+	constraintSet  [][]string
+}
+
+// NewIsolationFilter creates a filter that filters out stores with isolationLevel
+// For example, a region has 3 replicas in z1, z2 and z3 individually.
+// With isolationLevel = zone, if the region on z1 is down, we need to filter out z2 and z3
+// because these two zones already have one of the region's replicas on them.
+// We need to choose a store on z1 or z4 to place the new replica to meet the isolationLevel explicitly and forcibly.
+func NewIsolationFilter(scope, isolationLevel string, locationLabels []string, regionStores []*core.StoreInfo) Filter {
+	isolationFilter := &isolationFilter{
+		scope:          scope,
+		locationLabels: locationLabels,
+		constraintSet:  make([][]string, 0),
+	}
+	if len(isolationLevel) <= 0 || len(locationLabels) <= 0 {
```

Comment:
In this case, I think we don't need to create a filter?

Reply:
You are right. This part may should be moved out to where we create a filter.

Reply:
Can we remove it now?